src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingOperator.cs

```diff
@@ -114,6 +114,15 @@ public override BoundNode VisitNullCoalescingOperator(BoundNullCoalescingOperato
                 }
             }
 
+            // Optimize left ?? right to left.GetValueOrDefault() when left is T? and right is the default value of T
+            if (rewrittenLeft.Type.IsNullableType()
+                && rewrittenRight.IsDefaultValue()
```

Comment:
>rewrittenRight.IsDefaultValue() [](start = 19, length = 31)

Is this going to return true for the right side like ```new StructType()```? Please add a test for a scenario like that.

Reply:
It will, implicit value type constructors are already lowered to default expressions; I've added a test.